iamcaller/validate.go

```diff
@@ -0,0 +1,59 @@
+package iamcaller
+
+import (
+	"fmt"
+
+	"go.einride.tech/iam/iammember"
+	iamv1 "go.einride.tech/iam/proto/gen/einride/iam/v1"
+)
+
+// Validate checks that resolved caller info is valid.
+func Validate(caller *iamv1.Caller) error {
+	if err := validateMembers(caller); err != nil {
+		return fmt.Errorf("validate caller: %w", err)
+	}
+	if err := validateMetadata(caller); err != nil {
+		return fmt.Errorf("validate caller: %w", err)
+	}
+	return nil
+}
+
+func validateMembers(caller *iamv1.Caller) error {
+ValidateMembersLoop:
+	for _, member := range caller.Members {
+		// Validate the member.
+		if err := iammember.Validate(member); err != nil {
+			return err
+		}
+		// Validate that the member is present in the metadata.
+		for _, metadata := range caller.Metadata {
+			for _, metadataMember := range metadata.Members {
+				if member == metadataMember {
+					continue ValidateMembersLoop
+				}
+			}
+		}
```

Comment:
IMO move this to another function and remove the `ValidateMembersLoop` label.
Maybe `func hasMember(metadata *iamv1.Metadata, member string) bool`

Reply:
Good point! Fixing in a follow-up!